gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/oauth2/OAuth2SecurityController.groovy

```diff
@@ -52,10 +56,17 @@ class OAuth2SecurityController {
   @Autowired
   OAuth2ProtectedResourceDetails oAuth2ProtectedResourceDetails // OAuth2 config details
 
+  @Autowired
+  OAuth2Configuration oAuth2Configuration
+
   @Autowired(required = false)
   AnonymousSecurityConfig anonymousSecurityConfig
 
-  private String oauth2TokenValue
+  @Autowired
+  AnonymousAccountsService anonymousAccountsService
+
+  @Value('${services.deck.baseUrl:http://localhost/9000}')
```

Comment:
Good catch. Yes, this would redirect to deck's homepage each time, which then makes the call to '/oauth/info' again, which is successful the second time (returns the User object).